Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-programming/questions-programming.component.ts

```diff
@@ -1,9 +1,172 @@
+        this.getCategory();
+    }
+
+    /**
+     * Gets all the coding languages
+     */
+    private getCodingLanguage() {
+        this.questionsService.getCodingLanguage().subscribe((response) => {
+            this.codingLanguageList = response;
+            this.codingLanguageList.sort();
+            this.isLanguageReady = true;
+        });
+    }
+
+    /**
+     * Gets all the categories
+     */
+    private getCategory() {
+        this.categoryService.getAllCategories().subscribe((response) => {
+            this.categoryList = response;
+            this.isCategoryReady = true;
+        });
+    }
+
+    /**
+     * Adds language to the selectedLanguageList
+     * @param language : language to add
+     */
+    selectLanguage(language: string) {
+        let index = this.selectedLanguageList.indexOf(language);
+        if (index === -1) {
+            this.selectedLanguageList.push(language);
+            this.codingLanguageList.splice(this.codingLanguageList.indexOf(language), 1);
+        }
+        this.checkIfNoLanguageSelected();
+    }
+
+    /**
+     * Removes language from selectedLanguageList 
+     * @param language : Language to remove
+     */
+    removeLanguage(language: string) {
+        this.selectedLanguageList.splice(this.selectedLanguageList.indexOf(language), 1);
+        this.codingLanguageList.push(language);
+        this.checkIfNoLanguageSelected();
+        this.codingLanguageList.sort();
+    }
+
+    /**
+     * Toggles nolanguageSelected according to selectedLanguageList 
+     */
+    private checkIfNoLanguageSelected() {
+        this.nolanguageSelected = this.selectedLanguageList.length === 0 ? true : false;
+    }
+
+    /**
+     * Adds category to the Question 
+     * @param category : category to be added
+     */
+    selectCategory(category: string) {
+        this.questionModel.question.categoryID = this.categoryList.find(x => x.categoryName === category).id;
+    }
+
+    /**
+     * Adds difficulty to the Question
+     * @param difficulty
+     */
+    selectDifficulty(difficulty: string) {
+        this.questionModel.question.difficultyLevel = DifficultyLevel[difficulty];
+    }
+
+    /**
+     * Opens snack bar
+     * @param message: message to display
+     * @param enableRouting: enable routing after snack bar dismissed
+     * @param routeTo: routing path 
+     */
+    private openSnackBar(message: string, enableRouting: boolean = false, routeTo: string = '') {
+        let snackBarAction = this.snackBar.open(message, 'Dismiss', {
+            duration: 3000
+        });
+
+        if (enableRouting) {
+            snackBarAction.afterDismissed().subscribe(() => {
+                this.router.navigate([routeTo]);
+            });
+        }
+    }
+
+    /**
+     * Sends post request to add code snippet question
+     * @param codeSnippetQuestionForm : code snippet form of type ngForm
+     */
+    submitForm(codeSnippetQuestionForm: NgForm) {
```

Comment:
method name is not proper. you can pass only form values instead of entire form object.